internal/libyaml/reader_test.go

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package libyaml
+
+import (
+	"testing"
+
+	"go.yaml.in/yaml/v4/internal/testutil/assert"
+)
+
+func TestReader(t *testing.T) {
+	RunTestCases(t, "reader_test.yaml", map[string]TestHandler{
+		"reader-set-error":          runReaderSetErrorTest,
+		"reader-determine-encoding": runReaderDetermineEncodingTest,
+		"reader-update-raw-buffer":  runReaderUpdateRawBufferTest,
+		"reader-update-buffer":      runReaderUpdateBufferTest,
+		"reader-panic":              runReaderPanicTest,
+	})
+}
+
+func runReaderSetErrorTest(t *testing.T, tc TestCase) {
+	t.Helper()
+
+	parser := NewParser()
+
+	// args should be [problem, offset, value]
+	problem := tc.Args[0].(string)
+	offset := tc.Args[1].(int)
+	value := tc.Args[2].(int)
```

Comment:
The Len of Args should be tested and anything else than 3 should stop processing further